packages/react/src/useHasOneForm.ts

```diff
@@ -0,0 +1,81 @@
+import { useAutoRelationship, useRelationshipContext } from "./auto/hooks/useAutoRelationship.js";
+import { useHasOneController } from "./auto/hooks/useHasOneController.js";
+import { getRecordAsOption, useOptionLabelForField } from "./auto/hooks/useRelatedModel.js";
+import type { OptionLabel } from "./auto/interfaces/AutoRelationshipInputProps.js";
+import { useFormContext } from "./useActionForm.js";
+import { get } from "./utils.js";
+
+import { useEffect, useState } from "react";
+
+export const useHasOneForm = (props: {
+  field: string;
+  primaryLabel?: OptionLabel;
+  secondaryLabel?: OptionLabel;
+  tertiaryLabel?: OptionLabel;
+}) => {
+  const { path, metadata } = useAutoRelationship({ field: props.field });
+  const {
+    setValue,
+    getValues,
+    formState: { defaultValues, submitCount, isSubmitSuccessful },
+  } = useFormContext();
+
+  const { record, relatedModelOptions } = useHasOneController(props);
```

Comment:
Do we use this other hook anywhere else? If not i think we should merge them

Reply:
I use it with Polaris. I changed PolarisAutoHasForm to use `useHasOneForm.ts`